src/main/java/taxi/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package taxi.service.impl;
+
+import java.util.Optional;
+import taxi.dao.DriverDao;
+import taxi.exception.AuthenticationException;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Driver;
+import taxi.service.AuthenticationService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverDao driverDao;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverDao.findDriverByLogin(login);
+        if (driver.isEmpty()) {
+            throw new AuthenticationException("Login or password is incorrect");
+        }
+        if (driver.get().getPassword().equals(password)) {
+            return driver.get();
+        }
```

Comment:
![image](https://user-images.githubusercontent.com/117590814/214566044-990a4c3a-931e-4eba-99d3-4e4bd756e61b.png)
trust me, it's possible 😉

Reply:
It's not that hard, I just made it as it was in video and missed this part in task conditions)